### ISSUE TITLE
create mtg.train_model()

### ISSUE THREAD

**Issue by eeholmes** [SAFS-Varanasi-Internship/mindthegap#50]
Now create the mtg.train_model() function. Ask any needed clarifying questions

This along with mtg.set_up() are the 3 main user functions.

mtg.set_up() 

mtg.train_model()
* needs ds
* needs options
Does
* prepare_model_data()
* set up xbatcher
* train
* returns results object 

mtg.save_model_bundle(results) # current function but altered to get what it needs from results or maybe we change what a model bundle has. Let's deal with that after mtg.train_model() is created.

## Goal for results object

Make `mindthegap` easy to integrate with experiment trackers such as MLflow, DVC/DVCLive, or Weights & Biases without adding a hard dependency on any tracker. `Options` already records the configuration used for a run. We also need a standard way to expose how well the fitted model performed so that experiments can be compared.

## Proposed changes

Update the fit workflow so that:

```python
result = mtg.train_model()
```

returns a result object containing at least:

```python
result.options
result.metadata
result.metrics
result.history
result.model
```

### `result.metrics`

Provide a simple dictionary of final validation/evaluation metrics suitable for logging to an experiment tracker.

For example:

```python
{
    "val_loss": 0.31,
    "val_mae": 0.22,
}
```

These should come from the actual model evaluation/training results, not from `Options`.

Where appropriate, run an explicit post-fit evaluation such as:

```python
model.evaluate(..., return_dict=True)
```

rather than relying only on the last value in the Keras training history.

### `result.history`

Preserve the full epoch-by-epoch Keras training history, for example:

```python
{
    "loss": [...],
    "val_loss": [...],
    "mae": [...],
    "val_mae": [...],
}
```

This will allow experiment trackers to display training curves.

### Domain-specific metrics

Design `result.metrics` so additional gap-filling metrics can be added later, for example:

* MAE on synthetically masked validation pixels
* broadscale / multiscale error
* coastal vs. open-ocean performance
* other diagnostics used to compare gap-filling models

The result object should not assume that Keras `val_loss` is the only measure of model quality.

## Options serialization

Ensure `Options` has a stable serialization method suitable for experiment tracking, such as:

```python
options.to_dict()
options.to_flat_dict()
```

`to_flat_dict()` should produce tracker-friendly keys such as:

```python
{
    "target": "CHL",
    "gridder.method": "xbatcher",
    "gridder.tile_size": [128, 128],
    "fit.epochs": 50,
    "fit.batch_size": 32,
    "split.seed": 42,
}
```

## Keep tracker integration optional

Do **not** add MLflow, DVC, W&B, etc. as core dependencies or scatter tracker-specific calls throughout `mtg.fit()`.

The basic workflow should remain:

```python
result = mtg.fit(ds, options=options)
```

and external code should be able to do something like:

```python
mlflow.log_params(options.to_flat_dict())

result = mtg.fit(ds, options=options)

mlflow.log_metrics(result.metrics)
```

A generic tracker/callback interface for live epoch logging can be added later if needed.

## Provenance

Include useful resolved run metadata in `result.metadata`, including at least the `mindthegap` package version and other information already discovered during data/model preparation.

Keep this separate from `Options`:

* `Options` = requested run configuration
* `metadata` = resolved dataset/runtime information
* `metrics` = final model performance
* `history` = performance through training
* `model` = fitted model

**Comment by eeholmes** [comment]
completed with #52
